tests/testthat/test-as-data-frame.R

```diff
@@ -228,6 +224,25 @@ with_test_authentication({
         expect_true(is.data.frame(as.data.frame(ds, force=TRUE)))
     })
 
+    ds$hidden_var <- 1:20
+    ds <- hideVariables(ds, "hidden_var")
+
+    test_that("as.data.frame(force) retrieves hidden variables", {
+        skip_locally("Vagrant host doesn't serve files correctly")
+        expect_equal(hiddenVariables(ds), "hidden_var")
+
+        expect_warning(df <- as.data.frame(ds, force = TRUE),
+            "Variable hidden_var is hidden")
+        expect_equal(names(df), c("v1", "v2", "v3", "v4", "v5", "v6", "hidden_var"))
```

Comment:
My expectation was that requesting the whole dataset here would return only the visible variables.

Here's what I thought we decided on for behavior:
`as.data.frame.CrunchDataset` returns promises of everything, but with `force=TRUE|as.data.frame.CrunchDataFrame()` if there’s no filter, you never get hidden variables in the export, if you explicitly ask for hiddens you get all of them (if there’s no filter), and if there is a filter you get what you filtered if it’s hidden or not?

Reply:
Once you make this change, you'll probably want to duplicate this test with `include.hidden = TRUE` in `as.data.frame()` and confirm that you do get everything (including hiddens) when you request it that way.